vulnerabilities/data_source.py

```diff
@@ -56,6 +56,21 @@ def __post_init__(self):
         if not any([self.url, self.reference_id]):
             raise TypeError
 
+@dataclasses.dataclass
+class VulnPkgVersionRange:
+
+# This dataclass is nested inside the `Advisory` class below. It is used 
+# by the importers to transport vulnerable version ranges of a particular
+# package to the db insertion mechanism. 
+
+# The `purl` in  this case is a package url without a version. Eg value `pkg:maven/axis/axis`
+
+# The `version_ranges` contains a comma separated version  range expressions. Eg value `<4.1.4,>=4.0.0`
+# for which  the `purl` is vulnerable to some vulnerability. This `vulnerability` is specified by the parent `Advisory`
+# dataclass.
+
+    purl: str
+    version_ranges: str
```

Comment:
I called this a requirement in scancode https://github.com/nexB/scancode-toolkit/blob/ba4bbf216c6f44572662d16c76214a08b0a69e7e/src/packagedcode/models.py#L331
I wonder if we should rename it in scancode or here.... but in anycase I would like to keep in a single common name for the same concept.